passes/sat/expose.cc

```diff
@@ -474,12 +475,22 @@ struct ExposePass : public Pass {
 					}
 
 					if (flag_cut) {
-						RTLIL::Wire *in_wire = add_new_wire(module, w->name.str() + sep + "i", w->width);
-						in_wire->port_input = true;
-						out_to_in_map.add(sigmap(w), in_wire);
+						wire_map[w] = w->name.str() + sep + "i";
 					}
 				}
 			}
+			for (auto &wm : wire_map)
+			{
+				if (flag_input) {
+						RTLIL::Wire *in_wire = module->addWire(wm.second, GetSize(wm.first));
+						out_to_in_map.add(wm.first, in_wire);
+				}
+				if (flag_cut) {
```

Comment:
Are you sure the logic here is the same? Above there is a `if (flag_input) ... else ... if (flag_cut)` construct, this is just two independent conditions. I would probably use two wire maps to make this change obviously correct.

Reply:
Actually it can even be placed in blocks bellow, since we have flag_input or flag_cut they are mutually exclusive.

Reply:
Ah ok if they are mutually exclusive that's fine

Reply:
Do you think you can also make it an error to pass both `-cut` and `-input`? Right now one of the flags is silently ignored which isn't very nice.

Reply:
Sure